app/assets/javascripts/code/services/_sensors.js

```diff
@@ -164,17 +167,16 @@ export const findAvailableSensorsForParameter = (sort, sensors, parameter) =>
     sort(Object.values(sensors).filter(sensor => sensor.measurement_type === parameter.id));
 
 export const sort = sensors => {
-  const ORDERS = {
-    "Particulate Matter-AirBeam2-PM2.5 (µg/m³)": 4,
-    "Particulate Matter-AirBeam2-PM1 (µg/m³)": 3,
-    "Particulate Matter-AirBeam2-PM10 (µg/m³)": 2,
-    "Particulate Matter-AirBeam-PM (µg/m³)": 1,
-    "Humidity-AirBeam2-RH (%)": 2,
-    "Humidity-AirBeam-RH (%)": 1,
-    "Temperature-AirBeam2-F (F)": 2,
-    "Temperature-AirBeam-F (F)": 1,
-    "Sound Level-Phone Microphone (dB)": 1
-  };
+  const ORDERS = {};
+  ORDERS[buildSensorId({measurement_type: "Particulate Matter", sensor_name: "AirBeam2-PM2.5", unit_symbol: "µg/m³"})] = 4;
+  ORDERS[buildSensorId({measurement_type: "Particulate Matter", sensor_name: "AirBeam2-PM1", unit_symbol: "µg/m³"})] = 3;
+  ORDERS[buildSensorId({measurement_type: "Particulate Matter", sensor_name: "AirBeam2-PM10", unit_symbol: "µg/m³"})] = 2;
+  ORDERS[buildSensorId({measurement_type: "Particulate Matter", sensor_name: "AirBeam-PM", unit_symbol: "µg/m³"})] = 1;
+  ORDERS[buildSensorId({measurement_type: "Humidity", sensor_name: "AirBeam2-RH", unit_symbol: "%"})] = 2;
+  ORDERS[buildSensorId({measurement_type: "Humidity", sensor_name: "AirBeam-RH", unit_symbol: "%"})] = 1;
+  ORDERS[buildSensorId({measurement_type: "Temperature", sensor_name: "AirBeam2-F", unit_symbol: "F"})] = 2;
+  ORDERS[buildSensorId({measurement_type: "Temperature", sensor_name: "AirBeam-F", unit_symbol: "F"})] = 1;
+  ORDERS[buildSensorId({measurement_type: "Sound Level", sensor_name: "Phone Microphone", unit_symbol: "dB"})] = 1;
```

Comment:
Have you considered
```
{
   [buildSensorId({measurement_type: "Particulate Matter", sensor_name: "AirBeam2-PM2.5", unit_symbol: "µg/m³"})] = 4,
  ...
}
```
that way you don't have the mutation from `{}` to the filled one

Reply:
Great improvement even if it's a bit messy. Nice catch